codeview: Add keypress sounds

ptomato opened the pull request:
This adds sounds when typing space, equals, slash, and enter.

https://phabricator.endlessm.com/T24355

ptomato committed:
codeview: Add keypress sounds

This adds sounds when typing space, equals, slash, and enter.

https://phabricator.endlessm.com/T24355

ptomato committed:
codeview: Add sounds on delete

This adds one sound for deleting text with the Delete or Backspace keys
(we assume that's the case when deleting one character), and another
sound for deleting a larger selection of text.

https://phabricator.endlessm.com/T24355

cosimoc commented on src/codeview.js:
```diff
@@ -158,6 +164,27 @@ var Codeview = GObject.registerClass({
 
+    static _onInsertText(buffer, location, text) {
+        const sound = SoundServer.getDefault();
+        [...text].forEach(letter => {
```
I feel that you could simplify this by having a dictionary mapping letter -> sound name. Then you can look at `dictionary[letter]` and play that if it exists, without having to enumerate every case in the switch.

ptomato committed:
codeview: Add sounds for cut, copy, paste

These should work when pressing the key combos as well as selecting the
options from the right-click menu.

https://phabricator.endlessm.com/T24355
